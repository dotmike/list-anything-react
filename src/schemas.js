import { schema } from "normalizr";

export const subjectsSchema = new schema.Entity(
  "subjects",
  {},
  { idAttribute: "_id" }
);

const tabsSchema = new schema.Entity("tabs", {}, { idAttribute: "_id" });
const fieldsSchema = new schema.Entity("fields", {}, { idAttribute: "_id" });

export const subjectSchema = new schema.Entity(
  "subject",
  { tabs: [tabsSchema], fields: [fieldsSchema] },
  { idAttribute: "_id" }
);

export const subjectDataSchema = new schema.Entity(
  "subjectData",
  {},
  { idAttribute: "_id" }
);
