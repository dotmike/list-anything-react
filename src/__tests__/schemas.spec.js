import { subjectsSchema, subjectSchema, subjectDataSchema } from "../schemas";

describe("schemas", () => {
  it("subjectsSchema renders correctly", () => {
    expect(subjectsSchema).toMatchSnapshot();
  });

  it("subjectSchema renders correctly", () => {
    expect(subjectSchema).toMatchSnapshot();
  });

  it("subjectDataSchema renders correctly", () => {
    expect(subjectDataSchema).toMatchSnapshot();
  });
});
