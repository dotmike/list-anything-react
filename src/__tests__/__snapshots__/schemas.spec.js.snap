// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`schemas subjectDataSchema renders correctly 1`] = `
EntitySchema {
  "_getId": [Function],
  "_idAttribute": "_id",
  "_key": "subjectData",
  "_mergeStrategy": [Function],
  "_processStrategy": [Function],
  "schema": Object {},
}
`;

exports[`schemas subjectSchema renders correctly 1`] = `
EntitySchema {
  "_getId": [Function],
  "_idAttribute": "_id",
  "_key": "subject",
  "_mergeStrategy": [Function],
  "_processStrategy": [Function],
  "schema": Object {
    "fields": Array [
      EntitySchema {
        "_getId": [Function],
        "_idAttribute": "_id",
        "_key": "fields",
        "_mergeStrategy": [Function],
        "_processStrategy": [Function],
        "schema": Object {},
      },
    ],
    "tabs": Array [
      EntitySchema {
        "_getId": [Function],
        "_idAttribute": "_id",
        "_key": "tabs",
        "_mergeStrategy": [Function],
        "_processStrategy": [Function],
        "schema": Object {},
      },
    ],
  },
}
`;

exports[`schemas subjectsSchema renders correctly 1`] = `
EntitySchema {
  "_getId": [Function],
  "_idAttribute": "_id",
  "_key": "subjects",
  "_mergeStrategy": [Function],
  "_processStrategy": [Function],
  "schema": Object {},
}
`;
